libs/thermo/thermodynamics.py

```diff
@@ -93,7 +93,7 @@ def __init__(
             qgrau (np.array): Specific graupel content (kg/kg)
 
         """
-        self.temp = temp
-        self.rho = rho
-        self.press = press
-        self.massmix_ratios = [qvap, qcond, qice, qrain, qsnow, qgrau]
+        self.temp = deepcopy(temp)
```

Comment:
The assignment operator just creates a shallow copy of the array. This means it creates a new array with a references to the elements of the old one; so changing values in `temp` array, would trigger a change in `self.temp` as well. Which we assumed we do not want.

Reply:
ah I think here's the problem then because we do in fact what the microphysics scheme ti change the values for temp and massmix_ratios. I guess either you should return the updated values from the run function and re-assign the old ones, or do just a shallow copy.

Reply:
Oh wait I see you return `cp_thermo` at the end, so perhaps this isn't the problem afterall... hmmm